gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/impl/DefaultReactorHandlerRegistry.java

```diff
@@ -203,52 +200,55 @@ private Class<? extends Acceptor<?>> resolve(Class<? extends Acceptor> acceptor)
         );
     }
 
-    private void addAcceptors(List<? extends Acceptor<?>> newAcceptors) {
+    private void registerAcceptors(List<? extends Acceptor<?>> newAcceptors) {
         if (!newAcceptors.isEmpty()) {
             synchronized (this) {
                 newAcceptors.forEach(acceptor -> {
                     Class<? extends Acceptor<?>> acceptorType = resolve(acceptor.getClass());
-                    List registeredAcceptors = acceptors.get(acceptorType);
-                    if (registeredAcceptors == null) {
-                        registeredAcceptors = new ArrayList<>();
+                    if (acceptorType != null) {
+                        acceptors.compute(
+                            acceptorType,
+                            (k, v) -> {
+                                if (v == null) {
+                                    v = new ArrayList<>();
+                                }
+                                v.add(acceptor);
+                                // Sort list based on Acceptor comparable
+                                v.sort(null);
+                                return v;
+                            }
+                        );
                     }
-                    registeredAcceptors.add(acceptor);
-                    Collections.sort(registeredAcceptors);
-                    acceptors.put(acceptorType, registeredAcceptors);
                 });
             }
         }
     }
 
-    private void remove(List<Acceptor> previousAcceptors) {
+    private void unregisterAcceptors(List<? extends Acceptor<?>> previousAcceptors) {
         if (!previousAcceptors.isEmpty()) {
             synchronized (this) {
                 previousAcceptors.forEach(acceptor -> {
                     Class<? extends Acceptor<?>> acceptorType = resolve(acceptor.getClass());
-                    List registeredAcceptors = acceptors.get(acceptorType);
-                    if (registeredAcceptors != null) {
-                        registeredAcceptors.remove(acceptor);
-                        if (!registeredAcceptors.isEmpty()) {
-                            Collections.sort(registeredAcceptors);
-                            acceptors.put(acceptorType, registeredAcceptors);
-                        } else {
-                            acceptors.remove(acceptorType);
-                        }
+                    if (acceptorType != null) {
+                        acceptors.computeIfPresent(
+                            acceptorType,
+                            (k, v) -> {
+                                v.remove(acceptor);
+                                acceptor.clear();
+                                if (!v.isEmpty()) {
+                                    // Sort list based on Acceptor comparable
+                                    v.sort(null);
```

Comment:
same as above

Reply:
Cannot be used because of generic, but Collections.sort(X) does x.sort(null)

Reply:
![image](https://github.com/gravitee-io/gravitee-api-management/assets/4324887/bc157585-c087-4e0c-8f77-11c23f0e5924)